test/integ-new/main_new_test.go

```diff
@@ -192,67 +203,101 @@ func TestAPIAndPublicKeyService(t *testing.T) {
 func TestUIServiceHealthCheck(t *testing.T) {
 	t.Parallel()
 
-	addr := cfg.UIServiceAddr
-	wantStatusCode := http.StatusOK
 	healthCheckPath := "/health"
 
 	ctx := context.Background()
 
-	uri := addr + healthCheckPath
-	resp := testSendHTTPReq(ctx, t, uri, cfg.UIServiceIDToken)
-
-	defer resp.Body.Close()
+	uri := cfg.UIServiceAddr + healthCheckPath
 
-	if got, want := resp.StatusCode, wantStatusCode; got != want {
-		b, err := io.ReadAll(resp.Body)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Errorf("health check status code got=%d want=%d, response=%s", got, want, string(b))
-	}
+	testCallEndpoint(ctx, t, uri, cfg.UIServiceIDToken, http.StatusOK)
 }
 
-func TestCertRotatorService(t *testing.T) {
+func TestCertRotatorHealthCheck(t *testing.T) {
 	t.Parallel()
 
-	cases := []struct {
-		name           string
-		path           string
-		wantStatusCode int
-	}{
-		{
-			name:           "cert_rotator_health_check",
-			path:           "/health",
-			wantStatusCode: http.StatusOK,
-		},
-		{
-			name:           "cert_rotator_key_rotation",
-			path:           "/",
-			wantStatusCode: http.StatusOK,
-		},
-	}
+	healthCheckPath := "/health"
 
-	for _, tc := range cases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
+	ctx := context.Background()
 
-			ctx := context.Background()
+	uri := cfg.CertRotatorServiceAddr + healthCheckPath
 
-			uri := cfg.CertRotatorServiceAddr + tc.path
-			resp := testSendHTTPReq(ctx, t, uri, cfg.CertRotatorServiceIDToken)
+	testCallEndpoint(ctx, t, uri, cfg.CertRotatorServiceIDToken, http.StatusOK)
+}
 
-			defer resp.Body.Close()
+// Subtests must be run in sequence, and they have waits in between.
+// Therefore, they cannot be parallelized, and aren't a good fit for table testing.
+//
+//nolint:paralleltest
+func TestCertRotatorKeyRotation(t *testing.T) {
+	t.Parallel()
 
-			if got, want := resp.StatusCode, tc.wantStatusCode; got != want {
-				b, err := io.ReadAll(resp.Body)
-				if err != nil {
-					t.Fatal(err)
-				}
-				t.Errorf("Got unexpected status code, got=%d want=%d, response=%s", got, want, string(b))
-			}
-		})
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
```

Comment:
Why not `ctx := context.Background() `, we have been using everyelses, does it provide any additional benefit?